### PR TITLE
[XrdTls] Memory leak fixes in XrdTlsTempCA

### PR DESCRIPTION
**Pull request by jthiltges**
- Free XrdCryptoX509Chain when finished
- Duplicate and close file descriptors to free memory

### CODE REVIEW

File: src/XrdTls/XrdTlsTempCA.cc [hunk 1]
```diff
@@ -100,10 +100,10 @@ CASet::processFile(file_smart_ptr &fp, const std::string &fname)
     XrdCryptosslX509ParseFile(fp.get(), &chain, fname.c_str());
 
     auto ca = chain.Begin();
-    // Note we purposely leak the outputfp here; we are just borrowing the handle.
-    FILE *outputfp = fdopen(m_output_fd, "w");
-    if (!outputfp) {
+    file_smart_ptr outputfp(fdopen(dup(m_output_fd), "w"), &fclose);
```
Comment (bbockelm) [inline src/XrdTls/XrdTlsTempCA.cc:103]: Can you replace the direct `dup` call with the wrapper (which sets `O_CLOEXEC`): https://github.com/xrootd/xrootd/blob/master/src/XrdSys/XrdSysFD.hh#L54

**Comment by bbockelm** [comment]
@jthiltges - ran these through Valgrind.  All looks good!  I'd suggest to merge.

**Comment by abh3** [comment]
This will be merged after JT updates the pull request to use SysFD which hasn't happened yet. JT? Please be aware we are on a tight timeline for releasing 5.2 and it would be good to have this patch in that release but it won't happen until this change occurs.

File: src/XrdTls/XrdTlsTempCA.cc (hunk 1 above)
Comment (jthiltges) [reply src/XrdTls/XrdTlsTempCA.cc]: Good idea, thank you!

**Comment by jthiltges** [comment]
PR updated with Brian's suggestions and squashed. Thank you for your time and attention, and please let me know if there's any further changes needed.

**Pull request merged**